kolibri/core/content/management/commands/generate_fixtures.py

```diff
@@ -0,0 +1,516 @@
+            try:
+                del connections[db]
+            except AttributeError:
+                pass
+            call_command("migrate", interactive=False, database=db)
+
+    def handle(self, *args, **options):
+
+        app_name = options["app"]
+        file_name = options["file_name"]
+        generating_mode = options["mode"]
+        required_levels = options["levels"]
+
+        apps_names = ("content", "lessons", "exams", "logger")
+        modes_option = ("build_fixtures", "default_db")
+
+        if generating_mode not in modes_option:
+            print(
+                "\n invalid argument for mode, options are 'build_fixtures' and 'default_db' \n"
+            )
+            exit()
+
+        if app_name not in apps_names:
+            print("\n invalid app name \n")
+            exit()
+
+        # get app models names (content app in this case) that will be dumped
+        def get_app_models(app_name):
+
+            # first way, extracts only the base (main) models
+            models_names = [
+                f"{app_name}.{model.__name__}"
+                for model in list(apps.get_app_config(app_name).get_models())
+            ]
+
+            # second way, extracts all app models even if it doesn't explicilty inherit from models.Model (e.g. in content app : contentnode_has_prerequisite, contentnode_has_prerequisite,etc..)
+            all_models_names = [
+                f"{app_name}.{model}" for model in dict(apps.all_models[app_name])
+            ]
+
+            # which one should we use ? (i.e which models will be dumped ? the ones extraced through the first or the second way)
+
+            return models_names
+
+        # generated objects will be added here to get deleted later after dumping
+        generated_objects = []
+
+        def generate_random_id():
+            import uuid
+
+            return uuid.uuid4().hex
+
+        self.resource_Kind_iterator = 0
+
+        # some constants :
+        IGNORED_KINDS = ["quiz", "zim"]
+        ALL_RESOURCES_KINDS = [
+            kind.id
+            for kind in content_kinds.KINDLIST
+            if kind.id not in IGNORED_KINDS and kind.id != "topic"
+        ]
+        RESOURCES_COUNT = len(ALL_RESOURCES_KINDS)
+
+        FIXTURES_FILE_PATH = f"fixtures/{file_name}.json"
+
+        CHANNEL_DESCRIPTION = ""
+        LICENSE_NAME = "testing license"
+        LICENSE_NAME_DESCRIPTION = (
+            "ABC organization authorizes kolibri to use this these resources"
+        )
+        LICENSE_OWNER = "ABC org"
+        MIN_SCHEMA_VERSION = 1
+        DEVELOPER_NAME = "bedo khaled"
+
+        # for returning random choices
+        def choices(sequence, k):
+            return [random.choice(sequence) for _ in range(0, k)]
+
+        # puprpose: if we have a content node of certain kind what type of file (file_preset) should maps to this node  ?
+        content_kind_to_file_preset = {}
+        for format_object in format_presets.PRESETLIST:
```

Comment:
you can simplify this code creating previusly a list:
```
allowed_formats = list(set(format_presets.PRESETLIST).difference(IGNORED_KINDS))
for format_object in allowed_formats:
      if format_object.kind not in content_kind_to_file_preset:
          content_kind_to_file_preset[format_object.kind] = [format_object.id]
      else:
          content_kind_to_file_preset[format_object.kind].append(
              format_object.id
          )
```

Reply:
i didn't quiet understand your comment purpose, but the first line throw error 'unhashable type list'